spec/services/user_updater_spec.rb

```diff
@@ -249,6 +249,38 @@
       end
     end
 
+    context 'when sso overrides location' do
+      it 'does not change location' do
+        SiteSetting.sso_url = "https://www.example.com/sso"
+        SiteSetting.enable_sso = true
+        SiteSetting.sso_overrides_location = true
+
+        user = Fabricate(:user)
+        updater = UserUpdater.new(acting_user, user)
+
+        expect(updater.update(location: "new location")).to be_truthy
+
+        user.reload
+        expect(user.user_profile.location).not_to eq 'new location'
+      end
+    end
+
+    context 'when sso overrides website' do
+      it 'does not change website' do
+        SiteSetting.sso_url = "https://www.example.com/sso"
+        SiteSetting.enable_sso = true
+        SiteSetting.sso_overrides_website = true
+
+        user = Fabricate(:user)
+        updater = UserUpdater.new(acting_user, user)
+
+        expect(updater.update(website: "https://google.com")).to be_truthy
+
+        user.reload
+        expect(user.user_profile.website).not_to eq 'https://google.com'
+      end
+    end
+
```

Comment:
Something like this? Just following along what is done for the `sso_overrides_bio`.